src/main/java/storage/Storage.java

```diff
@@ -126,11 +96,35 @@ public AuthorList readAuthorListFromDataFile() {
             logger.info("Data restored");
             return authorList;
```

Comment:
In line 95, does gson implicitly know how to construct the `AuthorList` after providing it the `authorListType`? Since the only deserializer that is explicitly described is for `Author`. Just a personal question.

Reply:
I believe so. The default deserializer knows what to do. `Author` requires its own one as each `Manga` within its `MangaList` attribute requires a reference back to the `Author` to satisfy the bidirectional navigability we have between `Author` and `Manga`.